File: 118.pascals-triangle.js
/*
 * @lc app=leetcode id=118 lang=javascript
 *
 * [118] Pascal's Triangle
 */

// @lc code=start
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  const pascalTriangle = [];
  for (let i = 0; i < numRows; i++) {
    pascalTriangle[i] = [];
    pascalTriangle[i][0] = 1;
    for (let j = 1; j < i; j++) {
      pascalTriangle[i][j] =
        pascalTriangle[i - 1][j - 1] + pascalTriangle[i - 1][j];
    }
    pascalTriangle[i][i] = 1;
  }
  return pascalTriangle;
};
// @lc code=end
// 15/15 cases passed (48 ms)
// Your runtime beats 93.21 % of javascript submissions
// Your memory usage beats 7.69 % of javascript submissions (34 MB)

File: 118.pascals-triangle.java
import java.util.ArrayList;
import java.util.List;

/*
 * @lc app=leetcode id=118 lang=java
 *
 * [118] Pascal's Triangle
 */

// @lc code=start
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascalTriangle = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();
            row.add(1);
            for (int j = 1; j < i; j++) {
                int prevLeft = pascalTriangle.get(i - 1).get(j - 1);
                int prevRight = pascalTriangle.get(i - 1).get(j);
                row.add(prevLeft + prevRight);
            }
            row.add(1);
            pascalTriangle.add(row);
        }
        if (numRows > 0) {
            pascalTriangle.get(0).remove(0);
        }
        return pascalTriangle;
    }
}
// @lc code=end
// 15/15 cases passed (0 ms)
// Your runtime beats 100 % of java submissions
// Your memory usage beats 6.02 % of java submissions (37.1 MB)